### ISSUE TITLE
read_csv difference to pandas

### ISSUE THREAD

**Issue by kaiogu** [modin-project/modin#1008]
### System information
- **OS Platform and Distribution**: Ubuntu 18.04
- **Modin installed from (source or binary)**: pip
- **Modin version**: 0.6.3
- **Python version**: 3.7.3
- **Exact command to reproduce**: pd.read_csv(...)

### Describe the problem
```python
import pandas as pd
pd.read_csv(...)
```
The above works without a problem. But if I do the same with `import modin.pandas as pd`, I get an error.

This is apparently a problem in vanilla pandas too, if you set the parsing engine to C. But the standard pandas parsing engine is python, which gives a different result from standard modin.

### Source code / logs
I get the folowing error with my code if done under `import modin.pandas as pd`

```
pandas.errors.ParserError: Error tokenizing data. C error: EOF inside string starting at row 3677254
```

I can't post a raw stack trace because of company policy, but if it is really needed i can clean it up and post it.

**Comment by devin-petersohn** [comment]
Hi @kaiogu, thanks for posting!

Are you able to provide the data types of the columns (column names not needed)? I suspect there's an issue with how we split up the parsing.

If I can get the `dtypes` I should be able to get a better idea. In order to fix this, I need to be able to reproduce it.

**Comment by kaiogu** [comment]
Hi @devin-petersohn ,

If I get the dtypes after loading the data in vanilla pandas, I get:
```python
>>> print(data.dtypes.unique())
[dtype('O') dtype('float64') dtype('int64')]
```

I should mention that I also pass dtype and usecols extra arguments to `load_csv`:

```python
usecols = ['str_field_1',
           'float_field_1',
           'int_field_1']

dtype = {'str_field_1':object'
         'float_field_1':float,
         'int_field_1':int}

data = pd.read_csv("path/to/my/file.csv",
                   sep=';',
                   decimal=',',
                   usecols=usecols,
                   dtype=dtype)
```

Also, I get a warning about not having `psutil` when importing `modin.pd`:
```
WARNING: Not monitoring node memory since `psutil` is not installed. Install this with `pip install psutil` (or ray[debug]) to enable debugging of memory-related crashes.
```

Hope this, helps!

**Comment by devin-petersohn** [comment]
Thanks, this does help. The `psutil` warning is because it is an optional dependency of Ray, so generally that can be ignored.

With your information I was able to reproduce it, I will paste the logs and attach the file for our tracking.

I see the issue, and I will dig into this. It will be fixed for the next release, which is going out in the next day or so.

```
In [3]: df = pd.read_csv("newlines.csv", dtype={"col1":object, "col2":int, "col3":int, "col4":int})
                                                                                                     
---------------------------------------------------------------------------
RayTaskError(ParserError)                 Traceback (most recent call last)
<ipython-input-3-0e32a863d64d> in <module>
----> 1 df = pd.read_csv("newlines.csv")

~/software_builds/modin/modin/pandas/io.py in parser_func(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, squeeze, prefix, mangle_dupe_cols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, escapechar, comment, encoding, dialect, error_bad_lines, warn_bad_lines, skipfooter, doublequote, delim_whitespace, low_memory, memory_map, float_precision)
     95         if not kwargs.get("sep", sep):
     96             kwargs["sep"] = "\t"
---> 97         return _read(**kwargs)
     98 
     99     return parser_func

~/software_builds/modin/modin/pandas/io.py in _read(**kwargs)
    108         kwargs: Keyword arguments in pandas.read_csv
    109     """
--> 110     pd_obj = BaseFactory.read_csv(**kwargs)
    111     # This happens when `read_csv` returns a TextFileReader object for iterating through
    112     if isinstance(pd_obj, pandas.io.parsers.TextFileReader):

~/software_builds/modin/modin/data_management/factories.py in read_csv(cls, **kwargs)
     50     @classmethod
     51     def read_csv(cls, **kwargs):
---> 52         return cls._determine_engine()._read_csv(**kwargs)
     53 
     54     @classmethod

~/software_builds/modin/modin/data_management/factories.py in _read_csv(cls, **kwargs)
     54     @classmethod
     55     def _read_csv(cls, **kwargs):
---> 56         return cls.io_cls.read_csv(**kwargs)
     57 
     58     @classmethod

~/software_builds/modin/modin/engines/base/io/text/csv_reader.py in read(cls, filepath_or_buffer, **kwargs)
    162         # or based on the column(s) that were requested.
    163         if index_col is None:
--> 164             row_lengths = cls.materialize(index_ids)
    165             new_index = pandas.RangeIndex(sum(row_lengths))
    166             # pandas has a really weird edge case here.

~/software_builds/modin/modin/engines/ray/task_wrapper.py in materialize(cls, obj_id)
     18     @classmethod
     19     def materialize(cls, obj_id):
---> 20         return ray.get(obj_id)

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/ray/worker.py in get(object_ids, timeout)
   1455                     worker.core_worker.dump_object_store_memory_usage()
   1456                 if isinstance(value, RayTaskError):
-> 1457                     raise value.as_instanceof_cause()
   1458                 else:
   1459                     raise value

RayTaskError(ParserError): ray::modin.engines.ray.task_wrapper.deploy_ray_func() (pid=46436, ip=192.168.42.11)
  File "python/ray/_raylet.pyx", line 636, in ray._raylet.execute_task
  File "/Users/DevinPetersohn/software_builds/modin/modin/engines/ray/task_wrapper.py", line 8, in deploy_ray_func
    return func(**args)
  File "/Users/DevinPetersohn/software_builds/modin/modin/backends/pandas/parsers.py", line 86, in parse
    pandas_df = pandas.read_csv(BytesIO(to_read), **kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pandas/io/parsers.py", line 685, in parser_f
    return _read(filepath_or_buffer, kwds)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pandas/io/parsers.py", line 457, in _read
    parser = TextFileReader(fp_or_buf, **kwds)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pandas/io/parsers.py", line 895, in __init__
    self._make_engine(self.engine)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pandas/io/parsers.py", line 1135, in _make_engine
    self._engine = CParserWrapper(self.f, **self.options)
  File "/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/pandas/io/parsers.py", line 1917, in __init__
    self._reader = parsers.TextReader(src, **kwds)
  File "pandas/_libs/parsers.pyx", line 542, in pandas._libs.parsers.TextReader.__cinit__
  File "pandas/_libs/parsers.pyx", line 822, in pandas._libs.parsers.TextReader._get_header
  File "pandas/_libs/parsers.pyx", line 937, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas/_libs/parsers.pyx", line 2132, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: EOF inside string starting at row 0
```

[newlines.txt](https://github.com/modin-project/modin/files/4087091/newlines.txt)

**Comment by thenewera-ru** [comment]
I can confirm the problem on my side as well. Reading with vanilla pandas:
```python
import pandas as pd
data = pd.read_csv(
   PATH,
   quotechar='"', 
   skipinitialspace=True, 
   sep=',',
   header=0
)
```
gives no problem and correctly parses the data but with `import modin.pandas as mpd` I get the same `pandas.errors.ParserError: Error tokenizing data. C error: EOF inside string starting at row 1360`

**System Information:**
 - OS Platform and distribution: Mac OS Mojave `10.14.16`
 - Python version: `3.8.5`
 - Modin version: `0.9.1`
 - Ray version: `1.1.0`
 - Modin installed from: `pip`

**Comment by devin-petersohn** [comment]
@thenewera-ru Thanks for posting, can you open a new issue to track this?

Also, would you be able to try with the latest Modin release? We just released 0.10.0 so it would be good to understand if the issue is still happening.

**Comment by thenewera-ru** [comment]
> @thenewera-ru Thanks for posting, can you open a new issue to track this?
> 
> Also, would you be able to try with the latest Modin release? We just released 0.10.0 so it would be good to understand if the issue is still happening.

I've just tried upgrading `modin` to 0.10.0 as well as `ray` to 1.4.0 and the error still remains.
